File: app/src/main/java/com/example/jorgecaro/almacenamiento_jorge_caro/repositories/Local/IDao.java
package com.example.jorgecaro.almacenamiento_jorge_caro.repositories.Local;

import java.util.List;

/**
 * Created by jorge_caro on 3/14/18.
 */

public interface IDao<T> {
    boolean add(T entity);
    boolean delete(T entity);
    List<T> search();
    boolean checkIfExist(T entity);
    boolean update(T entity);
    List<T> search(int id);
}

File: app/src/main/java/com/example/jorgecaro/almacenamiento_jorge_caro/repositories/Local/IDao.kt
package com.example.jorgecaro.almacenamiento_jorge_caro.repositories.Local

/**
 * Created by jorge_caro on 3/14/18.
 */

interface IDao<T> {
    fun add(entity: T): Boolean
    fun delete(entity: T): Boolean
    fun search(): List<T>
    fun checkIfExist(entity: T): Boolean
    fun update(entity: T): Boolean
    fun search(id: Int): List<T>
    fun searchById(id: Int): T
}